File: src/main/java/nl/hanze/week1/assignment1/observers/BaseObserver.java
package nl.hanze.week1.assignment1.observers;

import nl.hanze.week1.assignment1.Stock;
import nl.hanze.week1.assignment1.View;

import java.util.ArrayList;
import java.util.Collection;

/**
 * Created by Laurens on 25-9-2017.
 */
public class BaseObserver {

    private Collection<View> views = new ArrayList<>();

    public void addView(View view) {
        views.add(view);
    }

    public void removeView(View view) {
        views.remove(view);
    }

    public void updateViews(Stock stock, double price) {
        views.forEach(view -> view.update(stock, price));
    }
}

File: src/main/java/nl/hanze/assignment1/observers/BaseObserver.kt
package nl.hanze.assignment1.observers

import nl.hanze.assignment1.StockUpdate
import nl.hanze.assignment1.View

/**
 * Created by Laurens on 25-9-2017.
 */
open class BaseObserver {

    private val views = mutableListOf<View>()

    fun addView(view: View) = views.add(view)

    fun removeView(view: View) = views.remove(view)

    fun updateViews(stockUpdate: StockUpdate) = views.forEach { it.update(stockUpdate) }
}
